test/test_networking.py

```diff
@@ -1274,6 +1364,72 @@ def test_legacy_server_connect_error(self):
             with pytest.raises(SSLError, match='testerror'):
                 ydl.urlopen('ssl://testerror')
 
+    def test_unsupported_impersonate_target(self):
```

Comment:
these tests are rough and a bit janky, but I still prefer having the tests compared to not.

blends in well with the rest of the file tbh :rofl: